lms/views/api/blackboard/exceptions.py

```diff
@@ -0,0 +1,10 @@
+class BlackboardGroupError(Exception):
+    def __init__(self, group_set):
+        self.details = {"group_set": group_set}
+        super().__init__(self.details)
+
+
+class BlackboardStudentNotInGroup(BlackboardGroupError):
+    """Student doesn't belong to any of the groups in a group set assignment."""
+
+    error_code = "blackboard_student_not_in_group"
```

Comment:
:+1: Looks good. There's an informal protocol being implemented here (the `error_code` and `details` attributes) that isn't documented anywhere or defined by an abstract class or anything, but this is used by the `api_error()` exception view that will catch this exception: https://github.com/hypothesis/lms/blob/72b7dd9e0eb60570936311c0cc6b1f8f224d2be4/lms/views/api/exceptions.py#L159-L200

(And actually I think the protocol is actually documented by the exception view's docstring.)

It's worth checking to make sure that `api_error()` actually *is* the exception view that handles this exception, I think it will be.

The `BlackboardGroupError` base class looks unnecessary to me, but I'm guessing you're expecting to add another subclass to it in future.